pixel/image.go

```diff
@@ -105,14 +139,15 @@ func (img Image[T]) setPixel(index int, c T) {
 	case zeroColor.BitsPerPixel() == 1:
 		// Monochrome.
 		x := index % int(img.width)
-		y := index / int(img.width)
-		offset := x + (y/8)*int(img.width)
+		offset := index / 8
+
 		ptr := (*byte)(unsafe.Add(img.data, offset))
 		if c != zeroColor {
-			*((*byte)(ptr)) |= 1 << uint8(y%8)
+			*((*byte)(ptr)) |= (1 << (7 - uint8(x%8)))
 		} else {
-			*((*byte)(ptr)) &^= 1 << uint8(y%8)
+			*((*byte)(ptr)) &^= (1 << (7 - uint8(x%8)))
```

Comment:
You are changing the format here! Specifically, this reverses the order of the bits in the `Monochrome` color type, thus breaking programs that were using `pixel.Monochrome` before with the old format.

If you want to add a new format, please do so by adding a new pixel type instead.

Reply:
I think I was the only one trying to use this format, and I did not complete that work (on Thumby) due to this code being incorrect in the current form. So I am pretty sure this is fixing something broken, not introducing a new format.

Reply:
Ok, seems reasonable.
Though we might want to add a test for this: just a small monochrome image (5 x 3 pixels or so - intentionally with a weird size) and check whether it is equivalent to a PNG image for example.